src/node.c

```diff
@@ -30,9 +30,6 @@ static bool S_can_contain(cmark_node *node, cmark_node *child) {
   if (node == NULL || child == NULL) {
     return false;
   }
-  if (NODE_MEM(node) != NODE_MEM(child)) {
```

Comment:
I'm not really complaining about this pull request, it's now rebased and working, what I do find very peculiar is the introduction of this new requirement in the API, that nodes can only be added to an AST with the same "allocator". This (already pretty old) paper might interest you: 
https://people.cs.umass.edu/~emery/pubs/berger-oopsla2002.pdf